### ISSUE TITLE
fromEither

### ISSUE THREAD

**Issue by srid** [kowainik/relude#237]
Would this be useful?

```haskell
      fromEither :: (e -> a) -> Either e a -> a
      fromEither = flip either id
```

Example usage:

```haskell
let val = fromEither error someFuncThatReturnsEither
```

**Comment by chshersh** [comment]
@srid Thanks for the suggestion. I don't think that `fromEither` name is descriptive enough. There are several functions on Hackage that has the name `fromEither` and have a different type. And `relude` also has a few combinators for dealing with `Either`. I personally almost never needed a function with a type like `(e -> a) -> Either e a -> a` so I don't consider it common enough for inclusion into `relude`.

**Comment by srid** [comment]
I named it such because conceptually it is similar to `fromMaybe`. "Conceptually", because both functions pull the value out of the structure, while using whatever fallbacks (for Maybe, that is a default value; for Either, that is applying a function).

I looked for this function more than a few times, and would come across only `fromLeft` and `fromRight` (not exactly what I wanted of course) from relude which are similar, except they discard the error value.

I'm somewhat surprised that `fromEither` is not commonly asked for ...
